generators/dart.js

```diff
@@ -198,3 +205,63 @@ Blockly.Dart.scrub_ = function(block, code) {
   var nextCode = Blockly.Dart.blockToCode(nextBlock);
   return commentCode + code + nextCode;
 };
+
+/**
+ * Gets a property and adjusts the value (taking into account indexing).
+ * @param {Blockly.Block} block the block
+ * @param {string} atId the property ID of the element to get
+ * @param {number=} opt_delta value to add
+ * @param {boolean=} opt_negate whether to negate the value
+ * @param {number=} opt_order highest order acting on this value
+ * @return {string|number}
+ */
+Blockly.Dart.getAdjusted = function(block, atId, opt_delta, opt_negate,
+    opt_order) {
+  var delta = opt_delta || 0;
+  var order = opt_order || Blockly.Dart.ORDER_NONE;
+  if (Blockly.Dart.ONE_BASED_INDEXING) {
+    delta--;
+  }
+  var defaultAtIndex = (Blockly.Dart.ONE_BASED_INDEXING) ? '1' : '0';
+  if (delta) {
+    var at = Blockly.Dart.valueToCode(block, atId,
+            Blockly.Dart.ORDER_ADDITIVE) || defaultAtIndex;
+  } else if (opt_negate) {
+    var at = Blockly.Dart.valueToCode(block, atId,
+            Blockly.Dart.ORDER_UNARY_PREFIX) || defaultAtIndex;
+  } else {
+    var at = Blockly.Dart.valueToCode(block, atId, order)
+        || defaultAtIndex;
+  }
+
+  if (Blockly.isNumber(at)) {
+    // If the index is a naked number, adjust it right now.
+    at = parseInt(at) + delta;
```

Comment:
parseInt always needs the radix specified (10) or else it will guess.  Most of the time it guesses correctly, but sometimes it will decide the number is octal.